sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -818,3 +818,34 @@ def test_coef_shape_not_zero():
     est_no_intercept = Lasso(fit_intercept=False)
     est_no_intercept.fit(np.c_[np.ones(3)], np.ones(3))
     assert est_no_intercept.coef_.shape == (1,)
+
+
+def test_enet_coordinate_descent():
+    """Test that a warning is issued if model does not converge"""
+    clf = Lasso()
+    n_samples = 15500
+    n_features = 500
+    X = np.ones([n_samples, n_features]) * 1e50
+    y = np.ones([n_samples])
+    assert_warns(ConvergenceWarning, clf.fit, X, y)
+
+
+def test_enet_coordinate_descent_gram():
+    """Test that a warning is issued if model does not converge"""
+    clf = Lasso(precompute=True)
+    n_samples = 15500
+    n_features = 500
+    X = np.ones([n_samples, n_features]) * 1e50
+    y = np.ones([n_samples])
+    assert_warns(ConvergenceWarning, clf.fit, X, y)
+
+
+def test_enet_coordinate_descent_multi_task():
+    """Test that a warning is issued if model does not converge"""
+    clf = MultiTaskLasso()
+    n_samples = 15500
+    n_features = 500
+    n_classes = 2
+    X = np.ones([n_samples, n_features]) * 1e50
+    y = np.ones([n_samples, n_classes])
+    assert_warns(ConvergenceWarning, clf.fit, X, y)
```

Comment:
to test this use tiny data and set max_iter to a very tiny number. it will make testing faster.

besides this is already done in the estimator eg:

https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/linear_model/coordinate_descent.py#L486

why is it not enough for you? bug?

Reply:
For the cases identified in #10813, the estimator raises no warning. Essentially, there are numerical issues causing the duality gap and tolerance to be equal to zero - as such the warning won't be raised in the estimator.